src/HDF5.jl

```diff
@@ -1968,14 +1975,15 @@ function create_external(source::Union{File,Group}, source_relpath, target_filen
 end
 
 # error handling
-function hiding_errors(f)
-    error_stack = H5E_DEFAULT
-    # error_stack = ccall((:H5Eget_current_stack, libhdf5), hid_t, ())
-    old_func, old_client_data = h5e_get_auto(error_stack)
-    h5e_set_auto(error_stack, C_NULL, C_NULL)
-    res = f()
-    h5e_set_auto(error_stack, old_func, old_client_data)
-    return res
+function silence_errors(f::Function)
+    estack = H5E_DEFAULT
+    func, client_data = h5e_get_auto(estack)
+    h5e_set_auto(estack, C_NULL, C_NULL)
+    try
```

Comment:
the `try`...`finally` statement is required in the case  where `f()` throws an `error` message, otherwise the error stack wouldn't get reset back.